third_party/envoy-gateway/Dockerfile

```diff
@@ -0,0 +1,19 @@
+# This is a copy of
+# https://github.com/envoyproxy/gateway/blob/v1.1.2/tools/docker/envoy-gateway/Dockerfile
+# but:
+# - using calico/base instead of distroless
+# - using TARGETARCH instead of TARGETPLATFORM.
+
+FROM --platform=linux/amd64 busybox AS source
+# Create the data directory for eg
+RUN mkdir -p /var/lib/eg
+
+FROM calico/base:latest-${TARGETARCH}
```

Comment:
OSS calico/base is parameterized in https://github.com/projectcalico/calico/pull/9678. Use the `CALICO_BASE` arg instead. There is no need to specify `TARGETARCH` because docker buildkit will handle it.